libs/blocks/global-navigation/global-navigation.js

```diff
@@ -373,6 +479,117 @@ class Gnav {
     decorationTimeout = setTimeout(decorateDropdown, CONFIG.delays.loadDelayed);
   };
 
+  decorateUniversalNav = async () => {
+    const config = getConfig();
+    const locale = getUniversalNavLocale(config.locale);
+    const environment = config.env.name === 'prod' ? 'prod' : 'stage';
+    const visitorGuid = window.alloy ? await window.alloy('getIdentity').then((data) => data?.identity?.ECID) : undefined;
+    const experienceName = getExperienceName();
+
+    const getDevice = () => {
+      const agent = navigator.userAgent;
+      for (const [os, osName] of Object.entries(osMap)) {
+        if (agent.includes(os)) return osName;
+      }
+      return 'linux';
+    };
+
+    await Promise.all([
+      loadScript(`https://${environment}.adobeccstatic.com/unav/1.0/UniversalNav.js`),
+      loadStyle(`https://${environment}.adobeccstatic.com/unav/1.0/UniversalNav.css`),
+    ]);
+
+    const getChildren = () => {
+      const children = [CONFIG.universalNav.components.profile];
+      // reset sign up value on change
+      children[0].attributes.isSignUpRequired = false;
+
+      this.universalNavComponents?.forEach((component) => {
+        if (component === 'profile') return;
+        if (component === 'signup') {
+          children[0].attributes.isSignUpRequired = true;
+          return;
+        }
+
+        children.push(CONFIG.universalNav.components[component]);
+      });
+
+      return children;
+    };
+
+    const onAnalyticsEvent = (data) => {
+      if (!data) return;
+
+      const getInteraction = () => {
+        const {
+          event: { type, subtype } = {},
+          source: { name } = {},
+          content: { name: contentName } = {},
+        } = data;
+
+        switch (`${name}|${type}|${subtype}|${contentName || ''}`) {
+          case 'profile|click|sign-in|':
+            return `Sign In|gnav|${experienceName}|unav`;
+          case 'profile|render|component|':
+            return `Account|gnav|${experienceName}`;
+          case 'profile|click|account|':
+            return `View Account|gnav|${experienceName}`;
+          case 'profile|click|sign-out|':
+            return `Sign Out|gnav|${experienceName}|unav`;
+          case 'app-switcher|render|component|':
+            return 'AppLauncher.appIconToggle';
+          case `app-switcher|click|app|${contentName}`:
+            return `AppLauncher.appClick.${convertToPascalCase(contentName)}`;
+          case 'app-switcher|click|footer|adobe-home':
+            return 'AppLauncher.adobe.com';
+          case 'app-switcher|click|footer|all-apps':
+            return 'AppLauncher.allapps';
+          case 'app-switcher|click|footer|adobe-dot-com':
+            return 'AppLauncher.adobe.com';
+          case 'app-switcher|click|footer|see-all-apps':
+            return 'AppLauncher.allapps';
+            // TODO: add support for notifications
```

Comment:
Is this a TODO we need to address before this goes to prod?

Reply:
We are waiting to get clear requirements on this. We will implement as soon as we have the code.